fire/cli/gnss/bernese.py

```diff
@@ -0,0 +1,378 @@
+    solution: BerneseSolution,
+) -> List[GNSSTidsserie]:
+    """
+    Opret tidsserier for punkter hvor en tidsserie af den givne type ikke
+    allerede eksisterer.
+    """
+    nye_tidsserier = []
+    for punkt in punkter:
+        tidsserier = [ts for ts in punkt.tidsserier if ts.srid == srid]
+        if not tidsserier:
+            tidsserie = GNSSTidsserie(
+                punkt=punkt,
+                srid=srid,
+                navn=f"{punkt.gnssnavn}_{tidsserietype}_{solution.datum}",
+                formål=f"GNSS-tidsserie for {punkt.gnssnavn}",
+                referenceramme=f"{solution.datum}",
+            )
+            nye_tidsserier.append(tidsserie)
+
+    return nye_tidsserier
+
+
+def opret_nye_gnss_observationer(
+    punkter: List[Punkt],
+    solution: BerneseSolution,
+    kovarianser: bool,
+) -> List[Observation]:
+    """
+    Opret GNSS observationer baseret på indholdet af den givne BerneseSolution.
+
+    ObservationsLængder oprettes altid, KoordinatKovarianser og ResidualKovarians
+    kun hvis `kovarianser` er True (== en COV-fil fra Bernese er tilgængelig).
+    """
+    nye_observationer = []
+    for punkt in punkter:
+        station = solution[punkt.gnssnavn]
+
+        observationslængde = station.obslængde.total_seconds() / 3600  # angives i timer
+
+        nye_observationer.append(
+            ObservationsLængde(
+                opstillingspunkt=punkt,
+                observationstidspunkt=solution.epoke,
+                varighed=observationslængde,
+            )
+        )
+
+        if not kovarianser:
+            continue
+
+        nye_observationer.append(
+            KoordinatKovarians(
+                opstillingspunkt=punkt,
+                observationstidspunkt=solution.epoke,
+                xx=station.kovarians.xx,
+                xy=station.kovarians.yx,
+                xz=station.kovarians.zx,
+                yy=station.kovarians.yy,
+                yz=station.kovarians.zy,
+                zz=station.kovarians.zz,
+            )
+        )
+
+        residualkovariansmatrix = station.dagsresidualer.kovarians_neu
+        nye_observationer.append(
+            ResidualKovarians(
+                opstillingspunkt=punkt,
+                observationstidspunkt=solution.epoke,
+                xx=residualkovariansmatrix[0][0],
+                xy=residualkovariansmatrix[0][1],
+                xz=residualkovariansmatrix[0][2],
+                yy=residualkovariansmatrix[1][1],
+                yz=residualkovariansmatrix[1][2],
+                zz=residualkovariansmatrix[2][2],
+            )
+        )
+
+    return nye_observationer
+
+
+def print_koordinat_tabel(koordinater: List[Koordinat]) -> None:
+    """
+    Skriv liste med nye koordinater til indsættelse i databasen.
+    """
+    koordinattabel = Table("Station", "T", "X", "Y", "Z", box=box.SIMPLE)
+
+    for k in koordinater:
+        koordinattabel.add_row(k.punkt.gnssnavn, str(k.t), str(k.x), str(k.y), str(k.z))
```

Comment:
Hvorfor `str(...)`? Er det for at undgå noget farve, som `rich` ellers viser tal med?

Reply:
rich brokker sig hvis ikke man spytter en streng ind. Jeg synes også det er kluntet...

Reply:
Ah, ja, det var uventet.